src/main/java/io/rpg/util/GameObjectFactory.java

```diff
@@ -0,0 +1,48 @@
+package io.rpg.util;
+
+import io.rpg.config.model.GameObjectConfig;
+import io.rpg.model.object.CollectibleGameObject;
+import io.rpg.model.object.DialogGameObject;
+import io.rpg.model.object.GameObject;
+import io.rpg.model.object.NavigationalGameObject;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Exposes collection of methods to create {@link io.rpg.model.object.GameObject} class instances.
+ */
+public class GameObjectFactory {
+  /**
+   * Creates {@link GameObject} instance basing on information contained in config.
+   *
+   * @param config description of object properties
+   * @return game object created based on information located in config
+   */
+  public static GameObject fromConfig(GameObjectConfig config) {
+    // TODO: add case for Player type here!
+    switch (GameObject.Type.valueOf(config.getTypeString())) {
```

Comment:
program się tu mi wywala, ale można szybko poprawić. wartości enumów są pisane wielkimi literami, a w plikach konfiguracyjnych jest małymi. valueOf zwraca wartości enumów na podstawie ich nazwy, musi się zgadzać dokładnie. jeśli w zamyśle było zwracanie na podstawie atrybutu asString, to trzeba napisać metodę w GameObject.Type do tego. oprócz tego w configu location-1 pojawia się typ 'interactive', trzeba zamienić raczej na 'dialog' i dodać wartości assetPath do wszystkich obiektów (bo zakładam, że obiekty są specyfikowane w plikach location, a katalogi objects są do wyrzucenia?)
```suggestion
    switch (GameObject.Type.valueOf(config.getTypeString().toUpperCase())) {
```

Reply:
[f4a0b7d](https://github.com/RPG-IO/rpg-io/pull/16/commits/f4a0b7d466be58fe51b7421623932deb6b282304)

Reply:
[b1b3031](https://github.com/RPG-IO/rpg-io/pull/16/commits/b1b3031cbb53b5232b270d96081a54e6274cdce5)

Reply:
Nie, katalogi objects maja zostać: będzie możliwość specyfikowania obiektów zarówno w konfiguracji lokacji jak i w katalogu obejcts -- zacząłem to robić ale wypadło. Dokończę to możliwie szybko.